src/app/users/ask-question/ask-question.component.ts

```diff
@@ -9,10 +26,101 @@ import { Post } from '../../models/Post';
 })
 export class AskQuestionComponent implements OnInit {
 
-  constructor() { }
+  form: FormGroup;
+  visible = true;
+  selectable = true;
+  removable = true;
+  addOnBlur = true;
+  separatorKeysCodes: number[] = [ENTER, COMMA];
+  tagCtrl = new FormControl();
+  filteredTags: Observable<string[]>;
+  tags: string[] = [];
+  allTags: string[] = ['Front-end', 'Back-end', 'Angular', 'HTML', 'CSS', 
+                        'JavaScript', 'Java', 'Spring','Database', 'SQL',
+                        'AWS', 'GCS', 'Azure'];
+
+  @ViewChild('tagInput', {static: false}) tagInput: ElementRef<HTMLInputElement>;
+  @ViewChild('auto', {static: false}) matAutocomplete: MatAutocomplete;
+
+  constructor(private fb: FormBuilder) {
+    this.filteredTags = this.tagCtrl.valueChanges.pipe(
+        startWith(null),
+        map((tag: string | null) => tag ? this._filter(tag) : this.allTags.slice()));
+    
+  }
+
+  add(event: MatChipInputEvent): void {
+    // Add tag only when MatAutocomplete is not open
+    // To make sure this does not conflict with OptionSelected Event
+    if (!this.matAutocomplete.isOpen) {
+      const input = event.input;
+      const value = event.value;
+
+      // Add our tag
+      if ((value || '').trim()) {
+        this.tags.push(value.trim());
+      }
+
+      // Reset the input value
+      if (input) {
+        input.value = '';
+      }
+
+      this.tagCtrl.setValue(null);
+    }
+  }
+
+  remove(tag: string): void {
+    const index = this.tags.indexOf(tag);
+
+    if (index >= 0) {
+      this.tags.splice(index, 1);
+    }
+  }
+
+  selected(event: MatAutocompleteSelectedEvent): void {
+    this.tags.push(event.option.viewValue);
+    this.tagInput.nativeElement.value = '';
+    this.tagCtrl.setValue(null);
+  }
+
+  private _filter(value: string): string[] {
+    const filterValue = value.toLowerCase();
+
+    return this.allTags.filter(tag => tag.toLowerCase().indexOf(filterValue) === 0);
+  }
+
+  questionInput: Object = {
+    "title": null,
+    "tags": null,
+    "question": null,
+  }
+
+  submitQuestion = function(event, qTitle, qTags, qQuestion) {
+    console.log('submit question reached');
+    event.preventDefault();
+    this.questionInput.title = qTitle;
+    this.questionInput.tags = this.tags;
+    this.questionInput.question = qQuestion;
+    console.log(this.questionInput);
+    console.log(this.tags);
+  }
```

Comment:
what does this do?  You're sending the stringified form.value below, but you're setting properties on this.questionInput here.  Its fine to explore while you're coding but this should be cleaned up